### Práctica guiada sobre carga de archivos con Pandas

Ejecute cada celda según indicaciones.

Lo primero es explorar la carpeta donde se esta ejecutando este cuaderno, alli debe estar la carpeta:
data/heart-disease-uci

In [6]:
import pandas as pd #Se importa la libreria, pd es el estandar

In [20]:
# Cargar un archivo CSV
pd.read_csv("data/heart-disease-uci/heart.csv")

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
0     63    1   3       145   233    1        0      150      0      2.3   
1     37    1   2       130   250    0        1      187      0      3.5   
2     41    0   1       130   204    0        0      172      0      1.4   
3     56    1   1       120   236    0        1      178      0      0.8   
4     57    0   0       120   354    0        1      163      1      0.6   
..   ...  ...  ..       ...   ...  ...      ...      ...    ...      ...   
298   57    0   0       140   241    0        1      123      1      0.2   
299   45    1   3       110   264    0        1      132      0      1.2   
300   68    1   0       144   193    1        1      141      0      3.4   
301   57    1   0       130   131    0        1      115      1      1.2   
302   57    0   1       130   236    0        0      174      0      0.0   

     slope  ca  thal target;;  
0        0   0     1      1;;  
1        0   0     2      1;;  
2        2   0     2      1;;  
3        2   0     2      1;;  
4        2   0     2      1;;  
..     ...  ..   ...      ...  
298      1   0     3      0;;  
299      1   0     3      0;;  
300      1   2     3      0;;  
301      1   1     3      0;;  
302      1   1     2      0;;  

[303 rows x 14 columns]

#### De la salida anterior podemos evidenciar que el archivo posee 303 filas y 14 columnas. Si es posible abralo en Excel y observe que el archivo NO tiene la columna que aqui se visualiza como 0,1,2... Esto se debe a que Pandas asigna por defecto indices a los DataFrames.

<hr>

Para iniciar el procesamiento de datos, importe el archivo "heart.csv" ubicado en la ruta data/heart-disease-uci/. Antes de manipular los datos, cree una nueva columna identificadora en la primera posición. Mantenga la celda A1 vacía para preservar los encabezados originales, e inserte en la celda A2 el identificador "Paciente 1", continuando secuencialmente con "Paciente 2", "Paciente 3", etc. para cada registro subsecuente. 

Al finalizar la modificación, exporte el conjunto de datos bajo el nombre "heart1.csv", asegurese de que el delimitador configurado sea la coma (,). En caso de que su sistema predetermine el punto y coma (;) como separador, especifique explícitamente el parámetro correspondiente durante el proceso de guardado para garantizar la compatibilidad con análisis posteriores.

In [37]:
pd.read_csv("data/heart-disease-uci/heart1.csv", index_col = 0) # Se cambia la columna de indexación

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  \
paciente_1     63    1   3       145   233    1        0      150      0   
paciente_2     37    1   2       130   250    0        1      187      0   
paciente_3     41    0   1       130   204    0        0      172      0   
paciente_4     56    1   1       120   236    0        1      178      0   
paciente_5     57    0   0       120   354    0        1      163      1   
...           ...  ...  ..       ...   ...  ...      ...      ...    ...   
paciente_299   57    0   0       140   241    0        1      123      1   
paciente_300   45    1   3       110   264    0        1      132      0   
paciente_301   68    1   0       144   193    1        1      141      0   
paciente_302   57    1   0       130   131    0        1      115      1   
paciente_303   57    0   1       130   236    0        0      174      0   

              oldpeak  slope  ca  thal  target    
paciente_1        2.3      0   0     1         1  
paciente_2        3.5      0   0     2         1  
paciente_3        1.4      2   0     2         1  
paciente_4        0.8      2   0     2         1  
paciente_5        0.6      2   0     2         1  
...               ...    ...  ..   ...       ...  
paciente_299      0.2      1   0     3         0  
paciente_300      1.2      1   0     3         0  
paciente_301      3.4      1   2     3         0  
paciente_302      1.2      1   1     3         0  
paciente_303      0.0      1   1     2         0  

[303 rows x 14 columns]

In [35]:
pd.read_excel("data/heart-disease-uci/heart.xlsx") #Leer archivos de Excel (xlxs, xls) | Arreglar el indice

Unnamed: 0  age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  \
0             2   63    1   3       145   233    1        0      150      0   
1             3   37    1   2       130   250    0        1      187      0   
2             4   41    0   1       130   204    0        0      172      0   
3             5   56    1   1       120   236    0        1      178      0   
4             6   57    0   0       120   354    0        1      163      1   
..          ...  ...  ...  ..       ...   ...  ...      ...      ...    ...   
298         300   57    0   0       140   241    0        1      123      1   
299         301   45    1   3       110   264    0        1      132      0   
300         302   68    1   0       144   193    1        1      141      0   
301         303   57    1   0       130   131    0        1      115      1   
302         304   57    0   1       130   236    0        0      174      0   

     oldpeak  slope  ca  thal  target  
0        2.3      0   0     1       1  
1        3.5      0   0     2       1  
2        1.4      2   0     2       1  
3        0.8      2   0     2       1  
4        0.6      2   0     2       1  
..       ...    ...  ..   ...     ...  
298      0.2      1   0     3       0  
299      1.2      1   0     3       0  
300      3.4      1   2     3       0  
301      1.2      1   1     3       0  
302      0.0      1   1     2       0  

[303 rows x 15 columns]

In [36]:
# Asegurandonos que el archivo anterior tiene los indices correctos
pd.read_excel("data/heart-disease-uci/heart.xlsx", index_col = 0)

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  \
paciente_1     63    1   3       145   233    1        0      150      0   
paciente_2     37    1   2       130   250    0        1      187      0   
paciente_3     41    0   1       130   204    0        0      172      0   
paciente_4     56    1   1       120   236    0        1      178      0   
paciente_5     57    0   0       120   354    0        1      163      1   
...           ...  ...  ..       ...   ...  ...      ...      ...    ...   
paciente_299   57    0   0       140   241    0        1      123      1   
paciente_300   45    1   3       110   264    0        1      132      0   
paciente_301   68    1   0       144   193    1        1      141      0   
paciente_302   57    1   0       130   131    0        1      115      1   
paciente_303   57    0   1       130   236    0        0      174      0   

              oldpeak  slope  ca  thal  target  
paciente_1        2.3      0   0     1       1  
paciente_2        3.5      0   0     2       1  
paciente_3        1.4      2   0     2       1  
paciente_4        0.8      2   0     2       1  
paciente_5        0.6      2   0     2       1  
...               ...    ...  ..   ...     ...  
paciente_299      0.2      1   0     3       0  
paciente_300      1.2      1   0     3       0  
paciente_301      3.4      1   2     3       0  
paciente_302      1.2      1   1     3       0  
paciente_303      0.0      1   1     2       0  

[303 rows x 14 columns]